catch/unit/compileAndCaptureOutput.py

```diff
@@ -0,0 +1,73 @@
+# Copyright (c) 2022 Advanced Micro Devices, Inc. All Rights Reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import os
+import subprocess
+import sys
+import unittest
+
+class CompileAndCapture(unittest.TestCase):
+  path = None
+  expected_error_count = None
+  file = None
+  error_string = None
+  platform = None
+
+  def setUp(self):
+    self.error_string = 'error:'
+    self.assertFalse(self.path == None)
+    self.assertFalse(self.file == None)
+    self.assertFalse(self.expected_error_count == None)
+    self.assertTrue(self.platform == 'amd' or self.platform == 'nvidia')
+
+  def test_atomic(self):
+    compiler_args = [
+      os.environ["HIP_PATH"] + '/bin/hipcc',
```

Comment:
We don't recommend setting the env var HIP_PATH when using HIP.  Also, the cmakefile has the logic to get the HIP installation being tested so the HIP path should be passed down to this script.

Reply:
The HIP_PATH environment variable is part of our local setup. It is possible that it shall not be used anymore and that it is obsolete, so the proposed approach is addressed and implemented within the following commit: https://github.com/ROCm-Developer-Tools/hip-tests/pull/233/commits/14cfa940797d1707b0e66f869026634589a059e8.
@scchan: Can you please review the changes?